test/test_torch.py

```diff
@@ -939,6 +944,385 @@ def test_arange(self):
+                # other order
+                fntensor_small_expanded = getattr(smallExpanded, fn)
+                fntensor_small_non_expanded = getattr(small, fn)
+                r1 = tensorfn(fntensor_small_expanded, largeExpanded, small2Expanded)
+                r2 = tensorfn(fntensor_small_non_expanded, large, small2)
+                self.assertEqual(r1, r2)
+                if fn in fns_3_args:
+                    fntensor_small2_expanded = getattr(small2Expanded, fn)
+                    fntensor_small2_non_expanded = getattr(small2, fn)
+                    r1 = tensorfn(fntensor_small2_expanded, smallExpanded, largeExpanded)
+                    r2 = tensorfn(fntensor_small2_non_expanded, small, large)
+                    self.assertEqual(r1, r2)
+                    r1 = tensorfn(fntensor_small2_expanded, largeExpanded, smallExpanded)
+                    r2 = tensorfn(fntensor_small2_non_expanded, large, small)
+                    self.assertEqual(r1, r2)
+
+            # now for torch. versions of functions
+            if fn not in fns_no_torch:
+                fntorch = getattr(torch, fn)
+
+                def torchfn(t1, t2, t3):
+                    if fn == "lerp":
+                        return fntorch(t1, t2, 0.5)
+                    elif fn in fns_3_args:
+                        return fntorch(t1, 1.0, t2, t3)
+                    else:
+                        return fntorch(t1, t2)
+                r1 = torchfn(large, small, small2)
+                r2 = torchfn(largeExpanded, smallExpanded, small2Expanded)
+                self.assertEqual(r1, r2)
+                # other order
+                r1 = torchfn(small, large, small2)
+                r2 = torchfn(smallExpanded, largeExpanded, small2Expanded)
+                self.assertEqual(r1, r2)
+                if fn in fns_3_args:
+                    r1 = torchfn(small2, small, large)
+                    r2 = torchfn(small2Expanded, smallExpanded, largeExpanded)
+                    self.assertEqual(r1, r2)
+                    r1 = torchfn(small2, large, small)
+                    r2 = torchfn(small2Expanded, largeExpanded, smallExpanded)
+                    self.assertEqual(r1, r2)
+
+            # now for in place functions
+            if fn not in fns_no_inplace and (fn not in fns_no_inplace_cuda or not largeExpanded.is_cuda):
+                # in-place tensor is not broadcastable; test only guaranteed
+                # to work by broadcasting other argument(s)
+
+                # need to clone largeExpanded so we can reuse, since functions are in-place
+                largeExpandedClone = largeExpanded.clone()
+
+                def tensorfn_inplace(t0, t1, t2=None):
+                    t0_fn = getattr(t0, fn + "_")
+                    if fn == "lerp":
+                        return t0_fn(t1, 0.5)
+                    elif fn == "masked_copy":
+                        return t0_fn(t1 < 0.5, t1.expand_as(t0))
```

Comment:
No, because it's not really masked_copy, more like masked scatter and the shape of the source has nothing really to do with the shape of the underlying tensor, so we don't broadcast it.

It's just unnecessarily confusing for me to have written the test to use the expanded/non-expanded tensors, though; I'll just use a 1-d arange of the proper length or something.